cmd/uptest/main.go

```diff
@@ -16,27 +16,21 @@
 package main
 
 import (
-	"log"
 	"os"
 	"path/filepath"
 	"strings"
-	"syscall"
 
 	"gopkg.in/alecthomas/kingpin.v2"
 
 	"github.com/upbound/uptest/internal"
 	"github.com/upbound/uptest/internal/config"
-	"github.com/upbound/uptest/internal/crdschema"
 )
 
 var (
 	app = kingpin.New("uptest", "Automated Test Tool for Upbound Official Providers").DefaultEnvars()
-	// e2e command
+	// e2e command (single command is preserved for backward compatibility)
+	// and we may have further commands in the future.
 	e2e = app.Command("e2e", "Run e2e tests for manifests by applying them to a control plane and waiting until a given condition is met.")
```

Comment:
nit: It seems that after removing the `crddiff` subcommands from uptest, we have only the `e2e` subcommand. What do you think about also removing this subcommand? And using the tool without any subcommand? As I remember correctly, before introducing the `crddiff` related subcommands, we do not have any subcommands in uptest.

I realize this is a breaking change, but before moving it, we can think in terms of simplification. The question that came to my mind is whether we will need different subcommands in the future.

I'm just writing this as a discussion point; it's not a big deal :)

Reply:
Hi @sergenyalcin,
Thanks for the discussion, very much appreciated.
I've also considered removing the sub-command and decided to keep it with the reasons given in the comment:
- Backwards compatibility: In addition to the official provider repositories, I think this tool is in use in other repositories we don't maintain.
- If we implement some other commands like the `preprocess` command we've been discussing for a while, it may make sense to add them as sub-commands to `uptest` as they would be coupled with the e2e tests (preprocessing for the e2e test manifests). If we want to add them as sub-commands, then having `e2e` as a sub-command makes sense again. If we break it today, then we will probably want to implement such related functionality in separate applications (binaries) in the future.

Considering these two points, I've decided to keep the `e2e` sub-command. Btw, I believe we had the `e2e` sub-command before `crddiff`. Please also see this [comment](https://github.com/upbound/uptest/pull/48#discussion_r1030824888).